packages/astro/src/frontend/dev-tools.ts

```diff
@@ -0,0 +1,152 @@
+                background-size: 28.28px 28.28px;
+              }
+
+              div[data-astro-hydration=undefined].astro-notify::after { 
+                background: #374151;
+                padding: 0.5rem;
+                content: "Static component. Use <"attr(data-astro-component-name)" client:load /> or another client:* directive to hydrate.";
+                left: 50%;
+                top: 50%;
+                transform: translate(-50%, -50%);
+              }
+
+              div[data-astro-hydration=undefined].astro-notify.show-notification::before { 
+                opacity: 0.7;
+              }
+
+              div[data-astro-hydration=undefined].astro-notify.show-notification::after { 
+                opacity: 1;
+              }
+
+              div[data-astro-hydration].astro-inspector::before {
+                transform: translateY(-100%);
+                content: "<"attr(data-astro-component-name)" "attr(data-astro-hydration-mode)" />"; 
+                position: absolute;
+                background: var(--dev-tool-hydration-enabled);
+                color: white;
+                padding: 4px;
+                border-radius: 4px 4px 0 0;
+                font-weight: 600;
+              }
+
+              div[data-astro-hydration=undefined].astro-inspector::before { 
+                background: var(--dev-tool-hydration-disabled);
+              }
+
+              div[data-astro-hydration].astro-inspector { border: 2px dashed var(--dev-tool-hydration-enabled); }
+              div[data-astro-hydration=undefined].astro-inspector { border: 2px dashed var(--dev-tool-hydration-disabled) }
+              `;
+
+/**
+ * Initialize devtools, use this instead of the automatic initalization
+ * at bottom of this file?
+ */
+export function initDevTools() {
+  addDevTools();
+  addHydrationMissingClickHandler();
+}
+
+/**
+ * Scan page for non-Astro components that are not being hydrated, and add
+ * click event handler to give visual feedback that component is static.
+ */
+function addHydrationMissingClickHandler() {
+  const allUnhydratedComponents = document.querySelectorAll('div[data-astro-hydration=undefined]');
+  allUnhydratedComponents.forEach((componentElement) => {
+    const componentEl = componentElement as HTMLElement;
+    const componentName = componentEl.dataset.astroComponentName;
+    componentEl.addEventListener('click', (e) => {
+      const elementHasFormParent = e.composedPath().filter((pathNode) => (pathNode as HTMLElement).tagName === 'FORM').length > 0;
+
+      if (!elementHasFormParent) {
+        componentElement.classList.add('show-notification');
+        console.warn(
+          `component < ${componentName} /> is being statically rendered. Use the client:* directives to hydrate it. Documentation: https://astro-docs-e7osusbql-pikapkg.vercel.app/core-concepts/component-hydration`
```

Comment:
probably want to update the url for the docs here to https://docs.astro.build, right?

Reply:
Good catch! Thanks for the input. I pushed a change for this.